Added test jrue_ols function

jruedai08 opened the pull request:
A new function called "jrue_ols" which don't depend on statsmodel now

jruedai08 committed:
Added test jrue_ols function

ligon reviewed (commented):
Nice first PR!  But why not make this a special case of GMM?

jruedai08 committed:
Added test jrue_ols function

jruedai08 committed:
(1) correct the code in gmm module, upload the function in gmm module, add data as the parameters, now most of the function in GMM module could run smoothly

(2) provide the document of GMM class, test the code in GMM module and compose it into of GMM_class in the document

(3) add some Unary Matrix Operator to allow more algebraic operations for dataframe

I have a question is that continuously_updated_gmm didn’t run the correct answer, its answer is too large

jruedai08 commented:
Modify the GMM module
(1) correct the code in gmm module, upload the function in gmm module, add data as the parameters, now most of the function in GMM module could run smoothly
(2) provide the document of GMM class, test the code in GMM module and compose it into of GMM_class in the document
(3) add some Unary Matrix Operator to allow more algebraic operations for dataframe
I have a question is that continuously_updated_gmm didn’t run the correct answer, its answer is too large

ligon commented on Metrics_Miscellany.egg-info/dependency_links.txt:
I don't think this file should be added to the repo.

ligon commented on Metrics_Miscellany.egg-info/top_level.txt:
I don't think this file should be added to the repo.

ligon commented on README.txt:
I fail to understand the reason for this file, with the one-line entry "README.org".

ligon commented on metrics_miscellany.org:
Not grasping the rationale for these new binary operators.  We already have __add__, etc.  What is the idea here?

jruedai08 replied:
The purpose of this operation is to enable algebraic operations on DataFrames. Although Pandas provides default operations, they typically require that columns have the same names in order to perform addition or subtraction. Otherwise, the result may contain missing values.

In cases like "Y - X @ beta," where column names differ, these operations become difficult to apply. My approach allows for algebraic operations on DataFrames with different column names, making it more flexible and easier to use in such scenarios.
<img width="541" alt="operation" src="https://github.com/user-attachments/assets/2aea7349-ed94-4374-ac61-01e71270119f">

ligon commented on metrics_miscellany.org:
I am not sure you're understanding the preferred semantics for gN & friends.  The "N" is mean to be an indication that the function is defined with respect to a given dataset.   The only place we need to explicitly refer to the data is where we explicitly use it, in whatever the gj function is.

jruedai08 replied:
Oh, maybe I misunderstood. So we only need to input the data in gj(), and not in other functions? Could you provide an example of how gj() should look like, or point me to where I can find it on your website? Really appreciate it!

ligon commented on metrics_miscellany.org:
```diff
@@ -1339,15 +1397,31 @@ __version__ = "0.3.1"
     gN.N,gN.k = e.shape
-    gN.N = e.count()  # Allows for possibility of missing data
+
+    if hasattr(e,'count'):
```
The function gj should be designed to return a pd.Series or pd.DataFrame, so if we don't have a count() method there's a deeper problem.  Maybe keep this assertion, and raise ValueError with a message to this effect?

jruedai08 replied:
Got it, I'll make the correction in that part.

ligon commented:
Jrue-

Great to hear from you. I've shared some comments on the PR. No time to check now, but is the continuously updated estimator off by a factor of N?

Thanks for your help on this,
-EL

On Sun Oct 13, 2024, 12:59 PM GMT, Zhu (Jrue) Dai ***@***.***> wrote:
> Modify the GMM module
> (1) correct the code in gmm module, upload the function in gmm module, add data as the parameters, now most of the function in GMM module could run smoothly
> (2) provide the document of GMM class, test the code in GMM module and compose it into of GMM_class in the document
> (3) add some Unary Matrix Operator to allow more algebraic operations for dataframe
> I have a question is that continuously_updated_gmm didn’t run the correct answer, its answer is too large
> —
> Reply to this email directly, view it on GitHub <https://github.com/ligon/Metrics_Miscellany/pull/1#issuecomment-2408970807>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAET2DFB5KZATA375KRVKO3Z3JVDZAVCNFSM6AAAAABHXAURWWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBYHE3TAOBQG4>.
> You are receiving this because you commented.Message ID: ***@***.***>

jruedai08 commented:
Hi Professor,
Got it, professor. I'll check that part later. Thanks for your reply! I'll
initiate another pull request after making the corrections.
Sincerely
Jrue

Ethan Ligon ***@***.***> 于2024年10月15日周二 07:29写道：

> Jrue-
>
> Great to hear from you. I've shared some comments on the PR. No time to
> check now, but is the continuously updated estimator off by a factor of N?
>
> Thanks for your help on this,
> -EL
>
> On Sun Oct 13, 2024, 12:59 PM GMT, Zhu (Jrue) Dai ***@***.***> wrote:
> > Modify the GMM module
> > (1) correct the code in gmm module, upload the function in gmm module,
> add data as the parameters, now most of the function in GMM module could
> run smoothly
> > (2) provide the document of GMM class, test the code in GMM module and
> compose it into of GMM_class in the document
> > (3) add some Unary Matrix Operator to allow more algebraic operations
> for dataframe
> > I have a question is that continuously_updated_gmm didn’t run the
> correct answer, its answer is too large
> > —
> > Reply to this email directly, view it on GitHub <
> https://github.com/ligon/Metrics_Miscellany/pull/1#issuecomment-2408970807>,
> or unsubscribe <
> https://github.com/notifications/unsubscribe-auth/AAET2DFB5KZATA375KRVKO3Z3JVDZAVCNFSM6AAAAABHXAURWWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBYHE3TAOBQG4>.
>
> > You are receiving this because you commented.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ligon/Metrics_Miscellany/pull/1#issuecomment-2412516379>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7AHOEMU5RY7VEJ63ROH73DZ3RHVFAVCNFSM6AAAAABHXAURWWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMJSGUYTMMZXHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
